core/test/unit/src/tasks/resolve-action.ts

```diff
@@ -47,6 +47,34 @@ describe("ResolveActionTask", () => {
     })
   }
 
+  describe("handling missing secrets in constructor", () => {
+    it("should throw if an action references missing secrets", async () => {
+      garden.setPartialActionConfigs([
+        {
+          kind: "Run",
+          type: "test",
+          name: "run-with-missing-secrets",
+          spec: {
+            command: ["echo", "${secrets.MISSING}"],
+          },
+        },
+      ])
+
+      expect(garden.secrets).to.be.empty
+      expect(garden.isLoggedIn()).to.be.false
+
+      const task = await getTask("Run", "run-with-missing-secrets")
+      expectError(() => getTask("Run", "run-with-missing-secrets"), {
+        contains: [
+          "The following secret names were referenced in configuration, but are missing from the secrets loaded remotely",
+          "Run run-with-missing-secrets: MISSING",
+          "No secrets have been loaded. If you have defined secrets for the current project and environment in Garden Cloud, this may indicate a problem with your configuration.",
```

Comment:
In this case we are clearly not logged in, so the error message is very confusing.

Better would be, if you are not logged in:
```suggestion
          "You are not logged in. Log in to get access to Secrets in Garden Cloud",
```

If you are logged in & loading secrets succeeded:
```suggestion
          "These secrets are missing in the project "<project name>" under environment "<env>" in Garden Cloud. Follow this link to manage secrets: <link>. Available secrets: xx, xx, xx",
```

Reply:
Thank you, good point!

I partially addressed these comments. Now we do print "You are not logged in. Log in to get access to Secrets in Garden Cloud." statement in the message footer if a user is not logged in.
The login status is propagated to the `throwOnMissingSecretKeys` and it's internal functions.

I did not propagated project config yet, that would be required to print the project and env info. Would it be better to do that in this PR?

Reply:
No, this is already way better than it was before 👍 That's amazing